ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -811,53 +811,94 @@ class TransactionService(session: LedgerSession) extends LedgerTestSuite(session
   test(
     "TXSingleMultiSame",
     "The same transaction should be served regardless of subscribing as one or multiple parties",
+    allocate(TwoParties)) {
+    case Participants(Participant(ledger, alice, bob)) =>
+      for {
+        _ <- ledger.create(alice, Dummy(alice))
+        _ <- ledger.create(bob, Dummy(bob))
+        aliceView <- ledger.flatTransactions(alice)
+        bobView <- ledger.flatTransactions(bob)
+        multiSubscriptionView <- ledger.flatTransactions(alice, bob)
+      } yield {
+        val jointView = aliceView ++ bobView
+        assertEquals(
+          "Single- and multi-party subscription yield different results",
+          jointView,
+          multiSubscriptionView)
+      }
+  }
+
+  test(
+    "TXSingleMultiSameTrees",
+    "The same transaction trees should be served regardless of subscribing as one or multiple parties",
+    allocate(TwoParties)
+  ) {
+    case Participants(Participant(ledger, alice, bob)) =>
+      for {
+        _ <- ledger.create(alice, Dummy(alice))
+        _ <- ledger.create(bob, Dummy(bob))
+        aliceView <- ledger.transactionTrees(alice)
+        bobView <- ledger.transactionTrees(bob)
+        multiSubscriptionView <- ledger.transactionTrees(alice, bob)
+      } yield {
+        val jointView = aliceView ++ bobView
+        assertEquals(
+          "Single- and multi-party subscription yield different results",
+          jointView,
+          multiSubscriptionView)
+      }
+  }
+
+  test(
+    "TXSingleMultiSameStakeholders",
+    "The same transaction should be served to all stakeholders",
     allocate(SingleParty, SingleParty)) {
     case Participants(Participant(alpha, alice), Participant(beta, bob)) =>
       for {
-        _ <- alpha.create(alice, Dummy(alice))
-        _ <- beta.create(bob, Dummy(bob))
-        aliceView <- alpha.flatTransactions(alice)
-        bobView <- beta.flatTransactions(bob)
-        _ <- synchronize(alpha, beta)
+        _ <- alpha.create(alice, AgreementFactory(bob, alice))
+        _ <- beta.create(bob, AgreementFactory(alice, bob))
         alphaView <- alpha.flatTransactions(alice, bob)
         betaView <- beta.flatTransactions(alice, bob)
       } yield {
-        val jointView = aliceView ++ bobView
+        // Strip command id to yield a transaction comparable across participant
+        def stripCommandId(transactions: Seq[Transaction]): Seq[Transaction] =
+          transactions.map(_.copy(commandId = ""))
```

Comment:
Maybe move this to a companion object (and make it private)?

Reply:
👍 submitted in https://github.com/digital-asset/daml/pull/3542